new_platforms/nuget/build/native/openenclave.targets

```diff
@@ -250,7 +275,11 @@
     <ItemDefinitionGroup Condition="'$(OEType)' != 'Enclave'">
         <ClCompile>
             <!-- OP-TEE Simulation -->
-            <PreprocessorDefinitions Condition="$(OEIsTz) == True And $(OEIsSim) == True">OE_SIMULATE_OPTEE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+            <PreprocessorDefinitions Condition="$(OEIsTz) == True And $(OEIsSim) == True And $(OEIsNewPlatforms) == True">OE_SIMULATE_OPTEE;OE_ENCLAVE_TYPE_AUTO=OE_ENCLAVE_TYPE_UNDEFINED;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+
+            <!-- Others -->
+            <PreprocessorDefinitions Condition="($(OEIsTz) == False Or $(OEIsSim) == False) And $(OEIsNewPlatforms) == True">OE_ENCLAVE_TYPE_AUTO=OE_ENCLAVE_TYPE_UNDEFINED;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+            <PreprocessorDefinitions Condition="($(OEIsTz) == False Or $(OEIsSim) == False) And $(OEIsLinux) == True And $(OEIsDebug) == True">_DEBUG;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

Comment:
Are the checks against `OEIsTz` and `OEIsSim` in lines 281 and 282 meaningful?

If targeting `new_platforms`, `OE_ENCLAVE_TYPE_AUTO` must be redefined regardless of `OEIsTz` and `OEIsSim`. Similarly, those two conditions bear no relationship to whether `_DEBUG` should be defined.

Reply:
Agree, removed those checks